src/assertions.js

```diff
@@ -167,24 +167,26 @@ const assertNoChildrenDeclaredYet = function(functionName, previousNode) {
 
 /**
  * Checks that a call to patchOuter actually patched the element.
- * @param {?Node} node The node requested to be patched.
- * @param {?Node} currentNode The currentNode after the patch.
+ * @param {?Node} startNode The value for the currentNode when the patch
+ *     started.
+ * @param {?Node} currentNode The currentNode when the patch finished.
+ * @param {?Node} expectedNextNode The Node that is expected to follow the
+ *    currentNode after the patch;
+ * @param {?Node} expectedPrevNode The Node that is expected to preceed the
+ *    currentNode after the patch.
  */
-const assertPatchElementNotEmpty = function(node, currentNode) {
-  if (node === currentNode) {
-    throw new Error('There must be exactly one top level call corresponding ' +
-        'to the patched element.');
-  }
-};
-
-
-/**
- * Checks that a call to patchOuter actually patched the element.
- * @param {?Node} node The node requested to be patched.
- * @param {?Node} previousNode The previousNode after the patch.
- */
-const assertPatchElementNoExtras = function(node, previousNode) {
-  if (node !== previousNode) {
+const assertPatchElementNoExtras = function(
+    startNode,
+    currentNode,
+    expectedNextNode,
+    expectedPrevNode) {
+  const wasUpdated = currentNode.nextSibling === expectedNextNode &&
```

Comment:
Wasn't quite sure how to do that, since you could have `elementOpen` calls in the subtree. Could store the count on the `currentParent`, but I that seems pretty messy too. Extra messy since you can `patchOuter` on a detached DOM node (but you can't remove it or replace it).